### PR TITLE
improvement: work with the new date/datetime type

### LINKED ISSUE #4968: Feature: Make parser suppport datatype with options like `DateTime(3)`/`DateTime(6)...`
**Summary**

After #4921 finished, we need to make `datetype` with options work, like `DateTime(3)/DateTime(6)` (`3` means precision, millisecond, `6` means microseconds).

For now, parser do not support datatype with options.

`DateTime(n)`    n between [ 0, 9]

cc @sundy-li @leiysky

### PR DESCRIPTION
**Pull request by Veeupup**
Signed-off-by: Veeupup <code@tanweime.com>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

* bump `sqlparser-rs` to the latest `49f4ec2`
* work with `datetime` work with precision

`TIMESTAMP` equals `DATETIME(0)`, with  second precision. If we want more precision, we can use `DATETIME(n)`, n ranges from 0~9

## Changelog

- Improvement

## Related Issues

Fixes #4968

### CODE REVIEW

File: tests/suites/0_stateless/02_function/02_0001_function_type_of.sql [hunk 1]
```diff
@@ -31,5 +31,5 @@ select 'UInt64 OP UInt64', typeof(65536 * 65536 + 65536 * 65536), typeof(65536 *
 
 select '=== TEST_datetimes';
 
-select typeof(now()) = 'DateTime_0';
+select typeof(now()) = 'DateTime(0)';
```
Comment (sundy-li) [inline tests/suites/0_stateless/02_function/02_0001_function_type_of.sql:34]: if it's zero, better to display `DateTime` ?

Comment (Veeupup) [reply tests/suites/0_stateless/02_function/02_0001_function_type_of.sql:34]: sounds reasonable

File: common/datavalues/src/types/type_datetime.rs [hunk 2]
```diff
@@ -36,10 +36,10 @@ pub struct DateTimeType {
     /// tz indicates the timezone, if it's None, it's UTC.
     tz: Option<String>,
 }
-const DATETIME_0: &str = "DateTime_0";
-const DATETIME_3: &str = "DateTime_3";
-const DATETIME_6: &str = "DateTime_6";
-const DATETIME_9: &str = "DateTime_9";
+const DATETIME_0: &str = "DateTime(0)";
```
Comment (sundy-li) [inline common/datavalues/src/types/type_datetime.rs:39]: Seems we can remove these variables

Comment (Veeupup) [reply common/datavalues/src/types/type_datetime.rs:39]: yes, should be removed,, will be fixed

**Comment by b41sh** [comment]
almost ok for me, please fix the test.

File: tests/suites/0_stateless/02_function/02_0001_function_type_of.sql (hunk 1 above)
Comment (Veeupup) [reply tests/suites/0_stateless/02_function/02_0001_function_type_of.sql]: resolved

File: common/datavalues/src/types/type_datetime.rs (hunk 2 above)
Comment (Veeupup) [reply common/datavalues/src/types/type_datetime.rs]: fixed

**Review by sundy-li — APPROVED** [review]

**Review by BohuTANG — APPROVED** [review]
:100:

**Comment by BohuTANG** [comment]
@sundy-li @Veeupup 

What's the databend final datetime data type is？
In some database systems(snow, [bigquery](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#timestamp_type), duckdb, [cockroachdb](https://www.cockroachlabs.com/docs/stable/timestamp.html)), they are introduce only one type: timestamp(or datetime), and the range:
```
0001-01-01 00:00:00 to 9999-12-31 23:59:59.999999 UTC
```
Supports precision levels from 0 (seconds) to 6 (microseconds) .

I suggest the default is TIMESTAMP(6), what do you think?

**Pull request merged**